mart/callbacks/adversarial_training.py

```diff
@@ -34,35 +48,54 @@ def setup(self, trainer, pl_module, stage=None):
     def teardown(self, trainer, pl_module, stage=None):
         pl_module.on_after_batch_transfer = self._on_after_batch_transfer
 
+    def wrap_model(self, model, dataloader_idx):
+        """Make a model, such that `output = model(batch)`."""
+
+        # Consume dataloader_idx
+        if hasattr(model, "attack_step"):
+
+            def model_forward(batch):
+                output = model.attack_step(batch, dataloader_idx)
+                return output
+
+        elif hasattr(model, "training_step"):
+            # Monkey-patch model.log to avoid spamming.
+            def model_forward(batch):
+                with MonkeyPatch(model, "log", lambda *args, **kwargs: None):
+                    output = model.training_step(batch, dataloader_idx)
+                return output
+
+        else:
+            model_forward = model
+
+        return model_forward
+
     def on_after_batch_transfer(self, pl_module, batch, dataloader_idx):
         batch = self._on_after_batch_transfer(batch, dataloader_idx)
 
-        # FIXME: Remove use of step
+        adversary = None
+
         trainer = pl_module.trainer
         if trainer.training:
             adversary = self.train_adversary
-            step = "training"
         elif trainer.validating:
             adversary = self.validation_adversary
-            step = "validation"
         elif trainer.testing:
             adversary = self.test_adversary
-            step = "test"
-        else:
-            return batch
```

Comment:
Why not just leave this instead of adding lines 86/87/89? I guess you could move the comment here.

Reply:
There is a chance that we are in one of the three phases training/validating/testing, but the adversary is not defined (None). So we need to know if adversary is None in lines 86/87/88

Reply:
Ah okay. Yeah I see now. Hope you have a test for this!

Reply:
I will add a trainer callback test in a separate PR.